### PR TITLE
Reorganizing repo

### PR DESCRIPTION
**Pull request by joswarmer**
Merged Federico's changes with mine and doing a new PR.

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -42,17 +42,19 @@ jobs:
         run: npm install
         
       - name: tsc
-        run: npm run build
+        run: npm run build -w packages
```
Comment (ftomassetti) [inline .github/workflows/main.yaml:45]: Wouldn't `npm run build` run the same, as it basically execute `npm run build -w packages`?

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -42,17 +42,19 @@ jobs:
...
-      - name: Create and iniytialize Postgres database
+      - name: Create and iniitialize Postgres database
         # Runs a script that creates a PostgreSQL table, populates
         # the table with data, and then retrieves the data.
-        run: node build/tools/database.js create && node build/tools/database.js init      
+        run: node packages/dbadmin/dist/tools/database.js create && node packages/dbadmin/dist/tools/database.js init      
       
       - name: start server and run tests (one step, else server will be down when tests start)
         run: |
+          cd packages/core
```
Comment (ftomassetti) [inline .github/workflows/main.yaml:54]: I think we do not need to step into the subdirectory, but we can specify in the top-level package.json the dev and test tasks as `npm run -w @lionweb/repository-core dev` and `npm run -w @lionweb/repository-dbadmin test`

File: README.md [hunk 3]
```diff
@@ -23,21 +24,50 @@ docker pull postgres:16.1
...
+npm run lint
+```
+
+### How to start the repository server
+Ensure that Postgress is running.
+The repository server is started with `npm run dev` in  the `packages/core` folder:
+
+```
+cd packages/core
+npm run dev
```
Comment (ftomassetti) [inline README.md:41]: See comments in the GitHub workflow

File: README.md [hunk 4]
```diff
@@ -23,21 +24,50 @@ docker pull postgres:16.1
...
+```
+
+### How to test
+Ensure the Postgres server and the rspository server are both running.
+Then do
+
+```
+npm run test
```
Comment (ftomassetti) [inline README.md:49]: See comments in the GitHub workflow

File: .github/workflows/main.yaml (hunk 2 above)
Comment (joswarmer) [reply .github/workflows/main.yaml:54]: Agreed, that is much more clear.

File: .github/workflows/main.yaml (hunk 1 above)
Comment (joswarmer) [reply .github/workflows/main.yaml:45]: Yes it does, will change it.

File: packages/core/package.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    "@types/mocha": "10.0.1",
+    "eslint": "8.56.0",
+    "chai": "4.3.10",
+    "mocha": "10.2.0",
+    "prettier": "3.1.0",
+    "ts-node": "10.9.2",
+    "typescript": "5.2.2"
+  },
+  "dependencies": {
+    "@lionweb/repository-dbadmin": "0.1.0",
```
Comment (ftomassetti) [inline packages/core/package.json:40]: I did not think you needed to explicitly add dependencies across sibling packages. I think for me it worked without adding them

File: packages/core/src/tmp.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
+insert into "lionweb_references"
```
Comment (ftomassetti) [inline packages/core/src/tmp.json:1]: Was this file committed by mistake?

File: packages/core/src/tmp2.json [hunk 7]
```diff
@@ -0,0 +1,467 @@
+{
```
Comment (ftomassetti) [inline packages/core/src/tmp2.json:1]: Was this file committed by mistake?

**Review by ftomassetti — APPROVED** [review]
Looks very good to me!

File: packages/core/package.json (hunk 5 above)
Comment (joswarmer) [reply packages/core/package.json:40]: They might work locally in the development environment, but when you install a _released_ package from npm the packages are standalone (no siblings anymore) and then you need to refer all used packages.
I dislike that the development environment does not flag this as an error, but that's the way it often is.

File: packages/core/src/tmp.json (hunk 6 above)
Comment (joswarmer) [reply packages/core/src/tmp.json:1]: Yes

File: packages/core/src/tmp2.json (hunk 7 above)
Comment (joswarmer) [reply packages/core/src/tmp2.json:1]: As well

File: README.md [hunk 8]
```diff
@@ -23,21 +24,50 @@ docker pull postgres:16.1
...
+### core
+The core repository
+
+### test
+Tests for the core package
 
 ## CI
-In GitHub actions a postgres server is started on a gost named `postres`.  In you local development environment this 
-hostname is also being used. You need to ensure that this hostname points to the postgres server. 
+In GitHub actions a postgres server is started on a host named `postres`.
```
Comment (dslmeinte) [inline README.md:71]: Or `postgres`?

File: packages/dbadmin/tsconfig.json [hunk 9]
```diff
@@ -0,0 +1,9 @@
+{
+  "extends": "../tsconfig.json",
+  "compilerOptions": {
+    "rootDir": "src",
+    "outDir": "dist",
+    "resolveJsonModule": true,                        /* Enable importing .json files */
```
Comment (dslmeinte) [inline packages/dbadmin/tsconfig.json:6]: I would expect this one then also in the top-level `tsconfig.json`.

File: README.md (hunk 8 above)
Comment (joswarmer) [reply README.md:71]: Hmmm ... ... you. might be right :-)

File: packages/dbadmin/tsconfig.json (hunk 9 above)
Comment (joswarmer) [reply packages/dbadmin/tsconfig.json:6]: you mean the resolveJsonModule ?

**Pull request merged**